internal/web/kubeclient/client.go

```diff
@@ -287,3 +300,121 @@ func (c *Client) filterNamespacesByAccess(ctx context.Context, namespaces []stri
 
 	return filteredNamespaces, false, nil
 }
+
+// CanPatchResource checks if the user has permission to patch a resource
+// by performing a SelfSubjectAccessReview with the "patch" verb.
+func (c *Client) CanPatchResource(ctx context.Context, group, resource, namespace string) (bool, error) {
+	kubeClient := c.GetClient(ctx)
+	if kubeClient == c.client {
+		// Privileged client has access to all resources.
+		return true, nil
+	}
+
+	ssar := &authzv1.SelfSubjectAccessReview{
+		Spec: authzv1.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authzv1.ResourceAttributes{
+				Verb:      "patch",
+				Group:     group,
+				Resource:  resource,
+				Namespace: namespace,
+			},
+		},
+	}
+
+	if err := kubeClient.Create(ctx, ssar); err != nil {
+		return false, fmt.Errorf("failed to create SelfSubjectAccessReview: %w", err)
+	}
+
+	return ssar.Status.Allowed, nil
+}
+
+// AnnotateResource annotates a resource with the provided map of annotations.
+func (c *Client) AnnotateResource(ctx context.Context, gvk schema.GroupVersionKind, name, namespace string, annotations map[string]string) error {
+	kubeClient := c.GetClient(ctx)
+
+	resource := &metav1.PartialObjectMetadata{}
+	resource.SetGroupVersionKind(gvk)
+
+	objectKey := client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+
+	if err := kubeClient.Get(ctx, objectKey, resource); err != nil {
+		return fmt.Errorf("unable to read %s/%s/%s: %w", gvk.Kind, namespace, name, err)
+	}
+
+	patch := client.MergeFrom(resource.DeepCopy())
+
+	existingAnnotations := resource.GetAnnotations()
+	if existingAnnotations == nil {
+		existingAnnotations = make(map[string]string)
+	}
+	maps.Copy(existingAnnotations, annotations)
+	resource.SetAnnotations(existingAnnotations)
+
+	if err := kubeClient.Patch(ctx, resource, patch); err != nil {
+		return fmt.Errorf("unable to annotate %s/%s/%s: %w", gvk.Kind, namespace, name, err)
+	}
+
+	return nil
+}
+
+// ToggleSuspension toggles the suspension of a Flux resource.
+// For Flux Operator resources, it uses the reconcile annotation.
+// For Flux resources, it patches the spec.suspend field.
+func (c *Client) ToggleSuspension(ctx context.Context, gvk schema.GroupVersionKind, name, namespace, requestTime string, suspend bool) error {
```

Comment:
Toggling is the action of inverting a boolean against its current state. I'd rename this function to `SetSuspension`.